common_helper_yara/yara_scan.py

```diff
@@ -1,33 +1,45 @@
-from subprocess import check_output, CalledProcessError, STDOUT
-import sys
-import re
 import json
 import logging
+import re
+from pathlib import Path
+from subprocess import check_output, CalledProcessError, STDOUT
+from typing import Optional, Any, Dict, Union
 
 from .common import convert_external_variables
 
 
-def scan(signature_path, file_path, external_variables={}, recursive=False):
+def scan(
+    signature_path: Union[str, Path],
+    file_path: Union[str, Path],
+    external_variables: Optional[Dict[str, Any]] = None,
```

Comment:
```suggestion
    external_variables: Dict[str, Any] = {},
```

Then you could later omit the assignment in case of `None`, couldn't you?

Reply:
Using mutible variables like lists or dicts as default is not allowed for good reasons (see e.g. https://docs.python-guide.org/writing/gotchas/#mutable-default-arguments)

Reply:
TIL. :-)